packages/xod-project/src/typeDeduction.js

```diff
@@ -192,26 +193,58 @@ const omitIntermediateAmbiguousPins = (
     )
   );
 
+// :: [a] -> Map a *
+const toMap = R.pipe(R.map(a => [a, true]), R.fromPairs);
```

Comment:
I'd add a comment explaining why this is needed.
Something along the lines of "Because R.has is faster than R.find"